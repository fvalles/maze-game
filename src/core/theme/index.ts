export { Fonts } from './fonts';
export { GlobalStyle } from './global-style';
