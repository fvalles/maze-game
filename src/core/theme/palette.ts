export const palette = {
  // color names are based on https://colornamer.netlify.app/
  artfulRed: '#B20600',
  crystalCut: '#F9F2ED',
  blackMarket: '#212240',
  janitor: '#2666CF',
  white: '#FFFFFF',
};
