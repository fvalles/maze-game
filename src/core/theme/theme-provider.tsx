import React, { FunctionComponent } from 'react';
import { DefaultTheme, ThemeProvider as Provider } from 'styled-components';
import { Fonts } from '.';

interface ThemeProviderProps {
  children?: React.ReactNode;
}

export const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  children,
}) => (
  <Provider
    theme={
      {
        Fonts,
      } as DefaultTheme
    }>
    {children}
  </Provider>
);
