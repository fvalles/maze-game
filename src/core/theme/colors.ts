import { palette } from './palette';

export const Colors = {
  // color names are based on https://colornamer.netlify.app/
  buttonBackground: palette.artfulRed,
  modalBackground: palette.crystalCut,
  restartButtonBackground: palette.janitor,
  text: palette.blackMarket,
  white: palette.white,
};

export type ColorsType = typeof Colors;
export type KeyColors = keyof ColorsType;
