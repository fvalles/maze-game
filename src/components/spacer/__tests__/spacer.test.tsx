import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Spacer } from '..';

describe('spacer component', () => {
  it('should render with default height', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Spacer />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('height', '32px');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom height of 64px', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Spacer size={64} />
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('height', '64px');
    expect(result).toMatchSnapshot();
  });
});
