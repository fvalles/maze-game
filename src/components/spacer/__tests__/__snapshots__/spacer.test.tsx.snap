// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacer component should render with custom height of 64px 1`] = `
.c0 {
  height: 64px;
}

<div
  className="c0"
  size={64}
/>
`;

exports[`spacer component should render with default height 1`] = `
.c0 {
  height: 32px;
}

<div
  className="c0"
/>
`;
